### PR TITLE
README: fix docs

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

Fix docs ahout `_curpkg_`

### CODE REVIEW

**Comment by lonng** [comment]
PTAL @kennytm @amyangfei

File: README.md [hunk 1]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
 
 ## Failpoint name best practice
 
-Because of all the failpoints name location in the same namespace, we need to be careful to avoid name conflict. There are
-some recommended naming rule to improve this situation.
+As you see above, `_curpkg_` will automatically to add around the original failpoint name in `failpoint.Eval` call.
```
Comment (amyangfei) [inline README.md]:
```suggestion
As you see above, `_curpkg_` will automatically wrap the original failpoint name in `failpoint.Eval` call.
```

File: README.md [hunk 2]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
...
-    failpoint.Inject("github.com/pingcap/tidb/planner/core/illegalPushDown", func(val failpoint.Value) {...})
-    failpoint.Inject("github.com/pingcap/pd/server/schedulers/balanceLeaderFailed", func(val failpoint.Value) {...})
-    ```
+func demo() {
+	// _curpkg_("the-original-failpoint-name") will be expand as `github.com/pingcap/tidb/ddl/the-original-failpoint-name`
+	if ok, val := failpoint.Eval(_curpkg_("the-original-failpoint-name")); ok {...}
+}
+```
+
+You do not need to care about `_curpkg_` in your application, which will automatically generate while run `failpoint-ctl enable`
```
Comment (amyangfei) [inline README.md]:
```suggestion
You do not need to care about `_curpkg_` in your application, which will be automatically generated while running `failpoint-ctl enable`
```

File: README.md [hunk 3]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
...
+and delete in `failpoint-ctl disable`.
+
+Because of all the failpoints name of a package location in the same namespace, we need to be careful to
+avoid name conflict. There are some recommended naming rule to improve this situation.
```
Comment (amyangfei) [inline README.md]:
```suggestion
avoid name conflict. There are some recommended naming rules to improve this situation.
```

File: README.md [hunk 4]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
 
 ## Failpoint name best practice
 
-Because of all the failpoints name location in the same namespace, we need to be careful to avoid name conflict. There are
-some recommended naming rule to improve this situation.
+As you see above, `_curpkg_` will automatically wrap the original failpoint name in `failpoint.Eval` call.
+You can think `_curpkg_` as a macro to automatically expand current package path to the failpoint name. For example,
```
Comment (kennytm) [inline README.md]:
```suggestion
You can think of `_curpkg_` as a macro that automatically prepends the current package path to the failpoint name. For example,
```

File: README.md [hunk 5]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
...
+package ddl // which parent package is `github.com/pingcap/tidb`
 
-    For examples,
-    ```go
-    failpoint.Inject("github.com/pingcap/tidb/ddl/renameTableErr", func(val failpoint.Value) {...})
-    failpoint.Inject("github.com/pingcap/tidb/planner/core/illegalPushDown", func(val failpoint.Value) {...})
-    failpoint.Inject("github.com/pingcap/pd/server/schedulers/balanceLeaderFailed", func(val failpoint.Value) {...})
-    ```
+func demo() {
+	// _curpkg_("the-original-failpoint-name") will be expand as `github.com/pingcap/tidb/ddl/the-original-failpoint-name`
```
Comment (kennytm) [inline README.md]:
```suggestion
	// _curpkg_("the-original-failpoint-name") will be expanded as `github.com/pingcap/tidb/ddl/the-original-failpoint-name`
```

File: README.md [hunk 6]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
...
+	if ok, val := failpoint.Eval(_curpkg_("the-original-failpoint-name")); ok {...}
+}
+```
+
+You do not need to care about `_curpkg_` in your application, which will be automatically generated while running `failpoint-ctl enable`
```
Comment (kennytm) [inline README.md]:
```suggestion
You do not need to care about `_curpkg_` in your application. It is automatically generated after running `failpoint-ctl enable`
```

File: README.md [hunk 7]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
...
+and delete in `failpoint-ctl disable`.
```
Comment (kennytm) [inline README.md]:
```suggestion
and is deleted with `failpoint-ctl disable`.
```

File: README.md [hunk 8]
```diff
@@ -407,19 +407,26 @@ instead of using failpoint marker functions.
...
+
+Because of all the failpoints name of a package location in the same namespace, we need to be careful to
```
Comment (kennytm) [inline README.md]:
```suggestion
Because all failpoints in a package share the same namespace, we need to be careful to
```

File: README.md [hunk 9]
```diff
@@ -431,7 +438,7 @@ some recommended naming rule to improve this situation.
 1. Define a group of marker functions
 2. Parse imports and prune a source file which does not import a failpoint
 3. Traverse AST to find marker function calls
-4. Marker function calls will be rewritten with an IF statement, which calls failpoint.Eval to determine whether a
+4. Marker function calls will be rewritten with an IF statement, which calls `failpoint.Eval `to determine whether a
```
Comment (kennytm) [inline README.md]:
```suggestion
4. Marker function calls will be rewritten with an IF statement, which calls `failpoint.Eval` to determine whether a
```

**Pull request merged**